src/test/java/touro/chess/BoardTest.java

```diff
@@ -0,0 +1,213 @@
+package touro.chess;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BoardTest {
+
+    private final Square[][] squares = new Square[8][8];
+    private final Board board = new Board();
+
+    public void createSquares(){
+        for(int column = 0; column < 8; column++) {
+            for (int row = 0; row < 8; row++) {
+                squares[column][row] = new Square(new Location(row, column));
+            }
+        }
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        com.sun.javafx.application.PlatformImpl.startup(() -> {
+        });
+    }
+
+    @Test
+    public void isLegal_validVerticalMove(){
+        //given
+        createSquares();
+
+        //when
+        Location currentLocation = new Location(2,1);
+        Location destinationLocation = new Location(4,1);
+        Square currentSquare = squares[2][1];
+        Square destinationSquare = squares[4][1];
+        Move move = new Move(currentLocation,destinationLocation,false);
+        currentSquare.setPiece(new RookPiece(currentLocation, PieceColor.White));
+        destinationSquare.setPiece(new RookPiece(destinationLocation, PieceColor.Black));
+
+        //then
+        Assert.assertTrue(board.isLegal(move));
```

Comment:
You shouldn't need `squares` since they are not the actual squares in the Board.
You shouldn't need RookPiece, since you don't care about specific pieces and should be using mocks.

Something like this.
```
Board board = new Board()
AbstractPiece piece = mock(AbstractPiece.class)
Location currentLocation = new Location(2,1);
Location destinationLocation = new Location(4,1);
board.setPiece(currentLocation, piece); // you need to write this
Move move = new Move(currentLocation, destinationLocation, false);
assertTrue(board.isLegal(move)
```